File: 3.cpp/C++/Projects/boost/python/ZmqSocket/ZmqSocket.cpp
#include "../../../Tools/ZmqSocket.hpp"
#include <boost/python.hpp>
using namespace boost::python;

BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(ZmqSocket_Init_overloads, Init, 3, 4);
BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS(ZmqSocket_Filter_overloads, Filter, 0, 1);
BOOST_PYTHON_MODULE(ZmqSocket)
{
    enum_<_CommProtocol>("_CommProtocol")
        .value("ZmqThread", ZmqThread)
        .value("ZmqProcess", ZmqProcess)
        .value("ZmqTcpIp", ZmqTcpIp);
    enum_<_MsgProtocol>("_MsgProtocol")
        .value("RepReq", RepReq)
        .value("PubSub", PubSub)
        .value("PushPull", PushPull);
    enum_<_ZmqType>("_ZmqType")
        .value("ZmqRep", ZmqRep)
        .value("ZmqReq", ZmqReq)
        .value("ZmqPub", ZmqPub)
        .value("ZmqSub", ZmqSub)
        .value("ZmqPush", ZmqPush)
        .value("ZmqPull", ZmqPull);
    enum_<_ZmqBindType>("_ZmqBindType")
        .value("ZmqBind", ZmqBind)
        .value("ZmqConnect", ZmqConnect);
    class_<_ZmqSocket>("_ZmqSocket", init<_CommProtocol, string, _ZmqType, optional<_ZmqBindType, int> >())
        .def("Init", &_ZmqSocket::Init, ZmqSocket_Init_overloads())
        .def("Filter", &_ZmqSocket::Filter, ZmqSocket_Filter_overloads())
        .def("Send", &_ZmqSocket::Send)
        .def("SendMore", &_ZmqSocket::SendMore)
        .def("Recv", &_ZmqSocket::Recv)
        .def("Dump", &_ZmqSocket::Dump);
}

File: 3.cpp/C++/Projects/boost/python/ZmqSocket/ZmqSocket.py
#!/usr/bin/env python

import ZmqSocket
import time

ZmqSock = ZmqSocket._ZmqSocket(ZmqSocket._CommProtocol.ZmqProcess,  #ZmqThread ZmqProcess ZmqTcpIp
                          "ipcexap.ipc",
                          ZmqSocket._ZmqType.ZmqRep,                #ZmqRep-ZmqReq ZmqPub-ZmqSub ZmqPush-ZmqPull
                          ZmqSocket._ZmqBindType.ZmqBind)           #ZmqBind-ZmqConnect

while 1 :
    time.sleep(1)
    ZmqSock.SendMore("aabbcc");
    ZmqSock.SendMore("ddeeff");
    ZmqSock.Send(" ");
    RecvMsg = ZmqSock.Recv()
    print(RecvMsg)

# SendTaskSock = ZmqSocket._ZmqSocket(ZmqSocket._CommProtocol.ZmqTcpIp,
#                           "192.168.1.21:6010",
#                           ZmqSocket._ZmqType.ZmqPush,
#                           ZmqSocket._ZmqBindType.ZmqBind)
# SendBegSock = ZmqSocket._ZmqSocket(ZmqSocket._CommProtocol.ZmqTcpIp,
#                           "192.168.1.21:6020",
#                           ZmqSocket._ZmqType.ZmqPush,
#                           ZmqSocket._ZmqBindType.ZmqConnect)
# SendBegSock.Send("10")                          
# i = 0
# j = 0
# while j < 10 :
#     time.sleep(1)
#     SendTaskSock.Send(str(i) + " " + str(i+1) + " " + str(i+2))
#     i = i+1
#     j = j+1

# RecvRsltSock = ZmqSocket._ZmqSocket(ZmqSocket._CommProtocol.ZmqTcpIp,
#                           "192.168.1.21:6020",
#                           ZmqSocket._ZmqType.ZmqPull,
#                           ZmqSocket._ZmqBindType.ZmqBind)
# RecvMsg = RecvRsltSock.Recv()
# TaskNum = int(RecvMsg)
# j = 0
# while j < TaskNum :
#     RecvMsg = RecvRsltSock.Recv()
#     print(RecvMsg)
#     j = j+1